base/operators.jl

```diff
@@ -328,8 +328,15 @@ eltype(::Type{Any}) = Any
 eltype(t::DataType) = eltype(supertype(t))
 eltype(x) = eltype(typeof(x))
 
-# function pipelining
+# function pipelining, composition & negation
+
 |>(x, f) = f(x)
+∘(f, g) = (x...)->f(g(x...))
+
+!(f::Function) = (x...)->!f(x...)
+^(f::Function, g::Function) = (x...)->f(x...) ^ g(x...)
+|(f::Function, g::Function) = (x...)->f(x...) || g(x...)
```

Comment:
I guess what @eschnett means is consistency in the sense that `f | g` means `x -> f(x) | g(x)` etc.